src/community/authkey/src/test/java/org/geoserver/security/AuthKeyAuthenticationTest.java

```diff
@@ -142,6 +143,43 @@ public void testMapperParameters() throws Exception {
         assertEquals("value2", fakeMapper.getMapperParameter("param2"));
     }
 
+    @Test
+    public void testMapperParametersFromEnv() throws Exception {
+        String authKeyUrlParam = "myAuthKeyParams";
+        String filterName = "testAuthKeyParams2";
+
+        AuthenticationKeyFilterConfig config = new AuthenticationKeyFilterConfig();
+        config.setAllowEnvParametrization(true);
```

Comment:
Following up with the above, the system variable can be set in a @Before and reset in a @After

Reply:
in theory, in practice it didn't work, I suppose because ALLOW_ENV_PARAMETRIZATION is statically initialized, do you know of another way to make it work?

Reply:
You're right, the system variable is static final... gah, not test friendly at all!
The variable should be made non final I believe, or you can try to force the test to run with a different classloader, see:

https://stackoverflow.com/questions/42102/using-different-classloaders-for-different-junit-tests

Also, I believe setting the variable should be done in a @BeforeClass to make that case work. @afabiani  ?

Reply:
@aaime @afabiani made a new commit, seems to be working this way